pkgs/development/python-modules/fmpy/default.nix

```
@@ -0,0 +1,213 @@
+{
+  lib,
+  stdenv,
+  fetchurl,
+  buildPythonPackage,
+  fetchFromGitHub,
+  pythonOlder,
+  cmake,
+  sundials,
+  lapack,
+  attrs,
+  lark,
+  lxml,
+  rpclib,
+  msgpack,
+  numpy,
+  scipy,
+  pytz,
+  dask,
+  requests,
+  matplotlib,
+  pyqtgraph,
+  notebook,
+  plotly,
+  hatchling,
+  pyside6,
+  jinja2,
+  flask,
+  dash,
+  dash-bootstrap-components,
+  qt6,
+  fmpy,
+  runCommand,
+  enableRemoting ? false,
+  versionCheckHook,
+  fmi-reference-fmus,
+}:
+
+let
+  # From sundials, build only the CVODE solver. C.f.
+  # src/native/build_cvode.py
+  cvode =
+    (sundials.overrideAttrs (prev: {
+      # hash copied from native/build_cvode.py
+      version = "5.3.0";
+      src = fetchurl {
+        url = "https://github.com/LLNL/sundials/releases/download/v5.3.0/sundials-5.3.0.tar.gz";
+        sha256 = "88dff7e11a366853d8afd5de05bf197a8129a804d9d4461fb64297f1ef89bca7";
+      };
+
+      cmakeFlags =
+        prev.cmakeFlags
+        ++ lib.mapAttrsToList (option: enable: lib.cmakeBool option enable) {
+          # only build the CVODE solver
+          BUILD_CVODE = true;
+
+          BUILD_CVODES = false;
+          BUILD_ARKODE = false;
+          BUILD_IDA = false;
+          BUILD_IDAS = false;
+          BUILD_KINSOL = false;
+        };
+    })).override
+      {
+        lapackSupport = false;
+        lapack.isILP64 = stdenv.hostPlatform.is64bit;
+        blas = lapack;
+        kluSupport = false;
+      };
+
+in
+buildPythonPackage rec {
+  pname = "fmpy";
+  version = "0.3.23";
+  disabled = pythonOlder "3.10";
+  pyproject = true;
+
+  # Bumping version? Make sure to look through the commit history for
+  # bumped native/build_cvode.py pins (see above).
+  src = fetchFromGitHub {
+    owner = "CATIA-Systems";
+    repo = "FMPy";
+    rev = "v${version}";
+    fetchSubmodules = true;
+    hash = "sha256-4jBYOymurGCbjT0WQjIRNsztwOXBYVTGLdc4kNSTOZw=";
+  };
+
+  nativeBuildInputs = [
+    cmake
+    hatchling
+  ];
+
+  propagatedBuildInputs = [
```

Use the modern Python attributes:

```suggestion
  dependencies= [
```